clusterctl/clusterdeployer/clusterclient.go

```diff
@@ -65,15 +67,16 @@ func (c *clusterClient) removeKubeconfigFile() error {
 	return os.Remove(c.kubeconfigFile)
```

Comment:
Does createTempFile handle removing the file automatically? I'm guessing not since we call os.Remove here. Which means that if there is an error creating the client we leak the file. We need to defer removing the file so that it gets cleaned up even on the error return paths.

Reply:
PR #390 merged and I have rebased this commit on top.